src/openrct2/object/TerrainEdgeObject.cpp

```diff
@@ -58,3 +60,10 @@ void TerrainEdgeObject::ReadJson(IReadObjectContext* context, json_t& root)
     PopulateTablesFromJson(context, root);
     NumImagesLoaded = GetImageTable().GetCount();
 }
+
+TerrainEdgeObject* TerrainEdgeObject::GetById(ObjectEntryIndex entryIndex)
+{
+    auto& objMgr = OpenRCT2::GetContext()->GetObjectManager();
+    auto obj = objMgr.GetLoadedObject(ObjectType::TerrainSurface, entryIndex);
```

Comment:
This should've used `ObjectType::TerrainEdge`

Reply:
Right, I guess this is (partly) responsible for the mapgen crashes?

Reply:
Partly indeed, the main issue was that the map generator would use surfaces and edges that are not necessarily available. The fallback to the first available edge in the list would now select a surface for the edge instead.